lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -627,6 +596,98 @@ function writeStringsToCtcFiles(locale, strings) {
   fs.writeFileSync(fullPath, JSON.stringify(output, null, 2) + '\n');
 }
 
+/**
+ * This function does three things:
+ *
+ *    - Add `meaning` property to ctc messages that have the same message so TC can disambiguate (otherwise it fails to import).
+ *    - Throw if the `meaning` of any collisions *also* collides (can't disambiguate messages).
+ *    - Throw if the known collisions has changed at all.
+ *
+ * @param {Record<string, CtcMessage>} strings
+ */
+function resolveMessageCollisions(strings) {
+  /** @type {Map<string, Array<[string, CtcMessage]>>} */
+  const stringsByMessage = new Map();
+
+  // Group all the strings by their message.
+  for (const entry of Object.entries(strings)) {
+    const collisions = stringsByMessage.get(entry[1].message) || [];
+    collisions.push(entry);
+    stringsByMessage.set(entry[1].message, collisions);
+  }
+
+  /** @type {Array<[string, CtcMessage]>} */
+  const allCollisions = [];
+  for (const group of stringsByMessage.values()) {
+    // If this message didn't collide with anything else, skip it.
+    if (group.length <= 1) continue;
+    allCollisions.push(...group);
+
+    // We have a message collision, time to check collisions on the `meaning` property.
+    /** @type {Map<string|undefined, Array<[string, CtcMessage]>>} */
+    const stringsByMeaning = new Map();
+    for (const [key, ctc] of group) {
+      ctc.meaning = ctc.description;
+
+      const collisions = stringsByMeaning.get(ctc.meaning) || [];
+      collisions.push([key, ctc]);
+      stringsByMeaning.set(ctc.meaning, collisions);
+    }
+
+    for (const meaningGroup of stringsByMeaning.values()) {
+      if (meaningGroup.length <= 1) continue;
+
+      const debugMeaningList = meaningGroup.map(entry => [entry[0], entry[1].meaning].join('\n'));
+      const debugCollisionsMessage = `${meaningGroup[0][1].message}\n\n${debugMeaningList.join('\n\n')}`;
+      throw new Error(`Each strings' \`message\` or \`description\` must be different for the translation pipeline. The following keys did not have unique \`meaning\` values:\n\n${debugCollisionsMessage}`);
+    }
+  }
+
+  // We survived fatal collisions, now check that the known collisions match our known list.
+  const collidingMessages = allCollisions.map(collision => collision[1].message).sort();
+
+  try {
+    expect(collidingMessages).toEqual([
+      '$MARKDOWN_SNIPPET_0$ elements do not have $MARKDOWN_SNIPPET_1$ text',
```

Comment:
Would `toMatchInlineSnapshot()` be useful?

Reply:
it would if this were a jest test, but we run it like a regular script so AFAIK `-u` isn't a thing we can do easily